quiz-api/data/quiz.go

```diff
@@ -0,0 +1,209 @@
+package data
+
+import (
+	"encoding/json"
+	"io"
+	"math"
+)
+
+// Number of questions for each quiz
+const NumOfQuestion = 5
+
+type Quiz []Question
+
+// Question is the structure to organize the quiz questions data
+type Question struct {
+	Text    string `json:"text"`
+	Answer  string `json:"-"`
+	Choices []Choice
+}
+
+// Choice contains the text and value of a given questions choice
+type Choice struct {
+	Text  string `json:"text"`
+	Value string `json:"value"`
+}
+
+// In-memory variable to keep records of the amount of correct answers per user
+var userAnswers = make(map[int]int, 5)
+
+// GetQuiz returns a Quiz with the questions from the system variable quiz
+func GetQuiz() Quiz {
+	return quiz
+}
+
+// ToJSON encodes a Quiz structure into JSON format
+func (q *Quiz) ToJSON(w io.Writer) error {
+	e := json.NewEncoder(w)
+	return e.Encode(q)
+}
+
+// VerifyAnswers returns the number of correct answers
+// It also increments the variables that serves as a record for the system
+func VerifyAnswers(answers []string) int {
+	// Returning variable with the number of questions answered correctly
+	var crtAns int
+	if len(answers) == 0 {
+		return -1
+	}

```

Comment:
Utilizar `error` no return. Exemplo:

```suggestion
	if len(answers) == 0 {
		return 0, errors.New("answers cant be empty")
	}
```

E a declaração da função passa a ser:
```golang
func VerifyAnswers(answers []string) (int, error)
```

Locais onde não há erro vc retorna o valor desejado e nil:
```golang
return number, nil
```